Lib/test/test_compileall.py

```diff
@@ -825,6 +1054,223 @@ def test_ignore_symlink_destination(self):
         self.assertTrue(os.path.isfile(allowed_bc))
         self.assertFalse(os.path.isfile(prohibited_bc))
 
+    def test_hardlink_deduplication_bad_args(self):
+        # Bad arguments combination, hardlink deduplication make sense
+        # only for more than one optimization level
+        self.assertRunNotOK(self.directory, "-o 1", "--hardlink_dupes")
+
+    def test_hardlink_deduplication_same_bytecode_all_opt(self):
```

Comment:
I'm not sure that it's useful to have a functional test on the CLI since we already have an unit test on compile_dir() function above.

Reply:
It seems that all the existing tests are also duplicated to test the imported module and also its CLI. What do you prefer? I'd keep at least `test_hardlink_deduplication_bad_args` and `test_hardlink_deduplication_same_bytecode_some_opt` in `CommandLineTestsBase`.